runtime/j9vm/jvm.c

```diff
@@ -5217,7 +5232,10 @@ JVM_SocketAvailable(jint descriptor, jint* result)
 #endif
 #if defined(J9UNIX) || defined(J9ZOS390)
 	if (0 <= descriptor) {
-		retVal = ioctl(descriptor, FIONREAD, result);
+		do {
+			retVal = ioctl(descriptor, FIONREAD, result);
```

Comment:
`ioctl` doesn't set `errno` to `EINTR`: http://man7.org/linux/man-pages/man2/ioctl.2.html.

Reply:
or http://pubs.opengroup.org/onlinepubs/9699919799/functions/ioctl.html